src/comparator/utils.rs

```diff
@@ -153,15 +154,20 @@ impl<'a> CompilationUnit<'a> {
 
         let args = self.cli_args(dependencies_artifacts);
 
-        InstrumentedCompiler::faked(
+        let mut compiler = InstrumentedCompiler::new(
             "glue".to_owned(),
             format!(
                 "extern crate {}; extern crate {};",
                 PREVIOUS_CRATE_NAME, NEXT_CRATE_NAME
             ),
-            args,
-        )?
-        .run()
+            test_compiler_settings(),
+        );
+
+        RunCompiler::new(args.as_slice(), &mut compiler)
+            .run()
+            .map_err(|_| anyhow!("Failed to compile crate"))?;
```

Comment:
Ideally we should not call the compiler via `RunCompiler::new`, but rather use the `run` method defined in the `Compiler` trait. The implementation of `run` for the instrumented compiler is also responsible for handling its internal state machine, which will allow us to mark finalize as private.

This is an attempt i made to unify the behaviour of every compiler we rely on.